pkg/importer/openapi.go

```diff
@@ -0,0 +1,317 @@
+		"date":    "date",
+	}
+	for swaggerName, syslName := range swaggerToSyslMappings {
+		l.types.Add(&ImportedBuiltInAlias{
+			name:   swaggerName,
+			Target: &SyslBuiltIn{syslName},
+		})
+	}
+	for name, schema := range l.spec.Components.Schemas {
+		if _, has := l.types.Find(name); !has {
+			_ = l.typeFromSchema(name, schema.Value)
+		}
+	}
+	l.types.Sort()
+}
+
+func (l *loader) typeFromRef(path string) Type {
+	path = strings.TrimPrefix(path, "#/components/schemas/")
+	if t, has := checkBuiltInTypes(path); has {
+		return t
+	}
+	if t, ok := l.types.Find(path); ok {
+		return t
+	}
+	if schema, has := l.spec.Components.Schemas[path]; has {
+		return l.typeFromSchema(path, schema.Value)
+	}
+	return nil
+}
+
+func (l *loader) typeFromSchemaRef(name string, ref *openapi3.SchemaRef) Type {
+	if ref == nil {
+		return nil
+	}
+	if t := l.typeFromRef(ref.Ref); t != nil {
+		return t
+	}
+	return l.typeFromSchema(name, ref.Value)
+}
+
+func sortProperties(props FieldList) {
+	sort.SliceStable(props, func(i, j int) bool {
+		return strings.Compare(props[i].Name, props[j].Name) < 0
+	})
+}
+
+func (l *loader) typeFromSchema(name string, schema *openapi3.Schema) Type {
+	for _, check := range []string{name, mapSwaggerTypeAndFormatToType(schema.Type, schema.Format, l.logger)} {
+		if t, found := l.types.Find(check); found {
+			return t
+		}
+	}
+	switch schema.Type {
+	case ObjectTypeName, "":
+		t := &StandardType{
+			name:       name,
+			Properties: FieldList{},
+		}
+		for pname, pschema := range schema.Properties {
+			var fieldType Type
+			if pschema.Value.Type == ArrayTypeName {
+				if atype := l.typeFromRef(pschema.Value.Items.Ref); atype != nil {
+					fieldType = &Array{Items: atype}
+				} else if atype := l.typeFromRef(pschema.Value.Items.Value.Type); atype != nil {
+					fieldType = &Array{Items: atype}
+				}
+			}
+			if fieldType == nil {
+				fieldType = l.typeFromSchemaRef(name+"_"+pname, pschema)
+			}
+			f := Field{
+				Name: pname,
+				Type: fieldType,
+			}
+			if !contains(pname, schema.Required) {
+				f.Optional = true
+			}
+			t.Properties = append(t.Properties, f)
+		}
+		sortProperties(t.Properties)
+		return l.types.AddAndRet(t)
+	case ArrayTypeName:
+		t := &Array{
+			name:  name,
+			Items: l.typeFromSchemaRef(name+"_obj", schema.Items),
+		}
+		if name != "" {
+			return l.types.AddAndRet(t)
+		}
+		return t
+	default:
+		baseType := mapSwaggerTypeAndFormatToType(schema.Type, schema.Format, l.logger)
+		if t, found := l.types.Find(baseType); found {
+			return t
+		}
+		if s, has := l.spec.Components.Schemas[schema.Type]; has {
+			return l.typeFromSchemaRef(schema.Type, s)
+		}
+
+		l.logger.Warnf("unknown schema.Type: %s", schema.Type)
+		return l.types.AddAndRet(NewStringAlias(name))
+	}
+}
+
+func (l *loader) initEndpoints() []MethodEndpoints {
+	epMap := map[string][]Endpoint{}
+
+	l.initGlobalParams()
+
+	for path, item := range l.spec.Paths {
+		ops := map[string]*openapi3.Operation{
```

Comment:
Are these the only operations supported in Open API, there is no Http OPTIONS and HEAD?

Reply:
If thesee end up being required they can be added later

Reply:
ok